libs/wire-subsystems/test/unit/Wire/MockInterpreters/Random.hs

```diff
@@ -0,0 +1,37 @@
+module Wire.MockInterpreters.Random where
+
+import Crypto.Random
+import Data.ByteString.Short (fromShort)
+import Data.Id
+import Imports
+import Polysemy
+import Polysemy.State
+import System.Random hiding (Random)
+import Wire.Sem.Random
+
+randomToStatefulStdGen :: (Member (State StdGen) r) => InterpreterFor Random r
+randomToStatefulStdGen = interpret $ \case
+  Bytes n -> do
+    fromShort <$> withStatefulGen (genShortByteString n)
+  Uuid -> withStatefulGen random
+  ScimTokenId -> Id <$> withStatefulGen random
+  LiftRandom m -> do
+    seedInt <- withStatefulGen (random @Int)
+    let seed = seedFromInteger $ toInteger seedInt
+        drg = drgNewSeed seed
+        (x, _) = withDRG drg m
+    pure x
+  NDigitNumber n -> withStatefulGen $ randomR (0, 10 ^ n - 1)
+
+runRandomPure :: InterpreterFor Random r
+runRandomPure = evalState defaulGen . randomToStatefulStdGen . raiseUnder
+
+defaulGen :: StdGen
+defaulGen = mkStdGen 0xBAD
+
+withStatefulGen :: (Member (State StdGen) r) => (StdGen -> (a, StdGen)) -> Sem r a
```

Comment:
This could be generalised to any type `s` instead of `StdGen`.

Reply:
I was wondering why there is no `modify`, but thought probably because of lack of atomicity guarantees. So I decided to write a specific one. So I didn't bother generalizing it in this module.

Reply:
What do you mean? There is a `modify`, but this one is an analogue of the `StateT` constructor. As a rule of thumb, I think it's good to turn types like `StdGen` above into type variables, because that limits the ways they could be used incorrectly in the implementation. Not a huge concern here though, obviously.